### PR TITLE
sfdnsres: Fix serialization issues

### PR DESCRIPTION
**Pull request by Thog**
Fix a crash on Monster Hunter Rise

### CODE REVIEW

File: Ryujinx.HLE/Utilities/StringUtils.cs [hunk 1]
```diff
@@ -60,6 +60,24 @@ public static byte[] HexToBytes(string hexString)
             return output;
         }
 
+        public static string ReadUtf8String(ReadOnlySpan<byte> data, out int dataRead)
+        {
+            dataRead = 0;
```
Comment (gdkchan) [inline Ryujinx.HLE/Utilities/StringUtils.cs:65]: You could just do `dataRead = data.IndexOf(0) + 1;` here. To account for the case of the span having no null terminator, you would need to additionally add `if (dataRead == 0) dataRead = data.Length`.

File: Ryujinx.HLE/HOS/Services/Sockets/Sfdnsres/Types/AddrInfo4.cs [hunk 2]
```diff
@@ -16,14 +17,34 @@ struct AddrInfo4
...
+        }
+
+        public void ToHostOrder()
+        {
+            Port = IPAddress.NetworkToHostOrder(Port);
+
+            RawIpv4AddressToSwapEndianess(ref Address);
+        }
+
+        public static void RawIpv4AddressToSwapEndianess(ref Array4<byte> address)
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Services/Sockets/Sfdnsres/Types/AddrInfo4.cs:42]: The name of this function seems a bit odd. `ToSwapEndianess`? I would call it `RawIpv4AddressNetworkEndianSwap`.

File: Ryujinx.HLE/Utilities/StringUtils.cs (hunk 1 above)
Comment (Thog) [reply Ryujinx.HLE/Utilities/StringUtils.cs:65]: No null terminator isn't a valid case here as official software use ``strlen``, will adjust anyway.

File: Ryujinx.HLE/Utilities/StringUtils.cs [hunk 3]
```diff
@@ -60,6 +60,19 @@ public static byte[] HexToBytes(string hexString)
             return output;
         }
 
+        public static string ReadUtf8String(ReadOnlySpan<byte> data, out int dataRead)
+        {
+            dataRead = data.IndexOf((byte)0) + 1;
+            int size = data.Length;
```
Comment (gdkchan) [inline Ryujinx.HLE/Utilities/StringUtils.cs:66]: Unused?

File: Ryujinx.HLE/Utilities/StringUtils.cs [hunk 4]
```diff
@@ -60,6 +60,19 @@ public static byte[] HexToBytes(string hexString)
...
+
+            if (dataRead <= 1)
```
Comment (gdkchan) [inline Ryujinx.HLE/Utilities/StringUtils.cs:68]: Why does it return null for an empty string?

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

**Review by AcK77 — APPROVED** [review]

**Pull request merged**